diffrax/_custom_types.py

```diff
@@ -53,34 +52,41 @@
 BufferDenseInfos = dict[str, PyTree[eqxi.MaybeBuffer[Shaped[Array, "times ..."]]]]
 sentinel: Any = eqxi.doc_repr(object(), "sentinel")
 
-LevyArea: TypeAlias = Literal["", "space-time"]
 
+class AbstractLevyArea(eqx.Module):
+    dt: eqx.AbstractVar[PyTree]
+    W: eqx.AbstractVar[PyTree]
 
-class LevyVal(eqx.Module):
+
+class TimeLevyArea(AbstractLevyArea):
+    dt: PyTree
+    W: PyTree
+
+
+class SpaceTimeLevyArea(AbstractLevyArea):
     dt: PyTree
     W: PyTree
-    H: Optional[PyTree]
-    bar_H: Optional[PyTree]
-    K: Optional[PyTree]
-    bar_K: Optional[PyTree]
+    H: PyTree
+    bar_H: Optional[PyTree] = None
+    K: Optional[PyTree] = None
+    bar_K: Optional[PyTree] = None
```

Comment:
@andyElking can you remind me, when are each of these values set?

Now that we explicitly distinguish at the type level whether or not the space-time Levy area is available, I'm thinking we should be able to remove the `Optional`, or move them to other subclasses of `AbstractLevyArea`?

Reply:
Ah, thanks!
In that case I think an internal type might make the most sense. It'd certainly help me wrap my head around the interface here :D

Reply:
Shall I remove the bar attributes from the `SpaceTimeLevyArea` in this pull then?

Reply:
Yes, probably best, but they need to be kept internally in VBT (for the purposes of levy_diff). If you need, I can make an edit to VBT tomorrow so that it doesn't use LevyVal internally, and you can freely reform it elsewhere.

Reply:
Yeah that would be great thanks.

Reply:
You can check PR #378